packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m

```diff
@@ -78,16 +78,28 @@ - (dispatch_queue_t)methodQueue
 
 EX_EXPORT_METHOD_AS(openBrowserAsync,
                     openBrowserAsync:(NSString *)authURL
+                    withArguments:(NSDictionary *)arguments
                     resolver:(EXPromiseResolveBlock)resolve
                     rejecter:(EXPromiseRejectBlock)reject)
 {
   if (![self initializeWebBrowserWithResolver:resolve andRejecter:reject]) {
     return;
   }
 
-  // Safari View Controller to authorize request
   NSURL *url = [[NSURL alloc] initWithString:authURL];
-  SFSafariViewController *safariVC = [[SFSafariViewController alloc] initWithURL:url entersReaderIfAvailable:NO];
+  SFSafariViewController *safariVC = NULL;
+  if(@available(iOS 11, *)) {
+    SFSafariViewControllerConfiguration *config = [[SFSafariViewControllerConfiguration alloc]init];
+    bool enabled = false;
+    NSString *collapseBarKey = @"enableBarCollapsing";
+    if([[arguments allKeys] containsObject:collapseBarKey]) {
+      enabled = [arguments[collapseBarKey] boolValue];
+    }
+    config.barCollapsingEnabled = enabled;
+    safariVC = [[SFSafariViewController alloc] initWithURL:url configuration:config];
+  }
+  // Safari View Controller to authorize request
+  safariVC = [[SFSafariViewController alloc] initWithURL:url];
```

Comment:
It looks as above `if` block is completely redundant because you're overriding `safariVC` here 🤔 	
Probably missing `else`?